spec/lock_mode_spec.rb

```diff
@@ -0,0 +1,124 @@
+require "spec_helper"
+
+RSpec.describe PgHaMigrations::LockMode do
+  describe "#<=>" do
+    {
+      access_share: {
+        above: %i[
+          row_share
+          row_exclusive
+          share_update_exclusive
+          share
+          share_row_exclusive
+          exclusive
+          access_exclusive
+        ],
+        below: [],
+      },
+      row_share: {
+        above: %i[
+          row_exclusive
+          share_update_exclusive
+          share
+          share_row_exclusive
+          exclusive
+          access_exclusive
+        ],
+        below: %i[access_share],
+      },
+      row_exclusive: {
+        above: %i[
+          share_update_exclusive
+          share
+          share_row_exclusive
+          exclusive
+          access_exclusive
+        ],
+        below: %i[
+          access_share
+          row_share
+        ],
+      },
+      share_update_exclusive: {
+        above: %i[
+          share
+          share_row_exclusive
+          exclusive
+          access_exclusive
+        ],
+        below: %i[
+          access_share
+          row_share
+          row_exclusive
+        ],
+      },
+      share: {
+        above: %i[
+          share_row_exclusive
+          exclusive
+          access_exclusive
+        ],
+        below: %i[
+          access_share
+          row_share
+          row_exclusive
+          share_update_exclusive
+        ],
+      },
+      share_row_exclusive: {
+        above: %i[
+          exclusive
+          access_exclusive
+        ],
+        below: %i[
+          access_share
+          row_share
+          row_exclusive
+          share_update_exclusive
+          share
+        ],
+      },
+      exclusive: {
+        above: %i[access_exclusive],
+        below: %i[
+          access_share
+          row_share
+          row_exclusive
+          share_update_exclusive
+          share
+          share_row_exclusive
+        ],
+      },
+      access_exclusive: {
+        above: [],
+        below: %i[
+          access_share
+          row_share
+          row_exclusive
+          share_update_exclusive
+          share
+          share_row_exclusive
+          exclusive
+        ],
+      },
+    }.each do |mode, other_modes|
```

Comment:
Creating this hash feels kind of dumb, but I kinda like how explicit it is

Reply:
I rewrote the tests to be even more explicit in https://github.com/braintree/pg_ha_migrations/pull/88/commits/e639e54fe2311f2eff38273636af96b023d488d1